src/MagicLogin.php

```diff
@@ -113,6 +115,58 @@ function (RegisterTemplateRootsEvent $event) {
             }
         );
 
+        Event::on(
+            UsersController::class,
+            UsersController::EVENT_BEFORE_ACTION,
+            function (ActionEvent $event) {
+                if (!$event->sender->action->actionMethod == 'actionSaveUser') {
+                    return;
+                }
+
+                // If we are updating an existing user then skip this.
+                $userId = $this->request->getBodyParam('userId');
+                if ($userId) {
+                    return;
+                }
+
+                $event->sender->requirePostRequest();
```

Comment:
This line could be moved above the "userId" parameter check because it is more of a general check which I would expect would potentially filter out more irrelevant requests.

Reply:
Moved in refactored.